classes/cache/slide_cache.php

```diff
@@ -141,8 +141,13 @@ public function load_for_cache($key) {
             $data['widthres'] = 0;
         } else if ($record->contenttype === 'image') {
             $imageinfo = $selectedfile->get_imageinfo();
-            $data['heightres'] = $imageinfo['height'];
-            $data['widthres'] = $imageinfo['width'];
+            if (!$imageinfo) {
+                $data['heightres'] = null;
+                $data['widthres'] = null;
+            } else {
+                $data['heightres'] = $imageinfo['height'];
+                $data['widthres'] = $imageinfo['width'];
+            }
```

Comment:
Seems fine, but how about this. If the image info returns `false`, we can treat it the same way we do if the selected file is not found which means the heightres and widthres are also zero by default, then we don't have to change anything in `block_carousel.php` and the filters can change to check against zeroes as well, unless you think block carousel needs to handle this extra null state?